web/src/components/pages/datasets/dataset-corpus-download.tsx

```diff
@@ -4,112 +4,133 @@ import {
-  const [sortedLocales] = useSortedLocales(
-    Object.keys(releaseData.locales),
-    getString
-  );
-  const [locale, setLocale] = useState(
-    releaseData.locales[globalLocale] ? globalLocale : 'en'
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedDataset, setSelectedDataset] = useState<LanguageDatasets>();
+  const [LanguageDatasets, setLanguageDatasets] = useState<LanguageDatasets[]>(
+    []
   );
+  const api = useAPI();
 
-  const bundleState = generateBundleState(locale, releaseId);
+  const [locale, setLocale] = useState(initialLanguage);
 
   const handleLangChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     const newLocale = event.target.value;
+
     setLocale(newLocale);
   };
 
-  const handleVersionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const newReleaseId = event.target.value;
-    setReleaseId(newReleaseId);
-  };
+  useEffect(() => {
+    setIsLoading(true);
 
+    api.getLanguageDatasetStats(locale).then(data => {
+      setLanguageDatasets(
+        data.filter(
+          (dataset: LanguageDatasets) =>
+            !!dataset.checksum && !!dataset.download_path
+        )
+      );
+      setSelectedDataset(data[0]);
+      setIsLoading(false);
+    });
+  }, [locale]);
   return (
     <div className="dataset-corpus-download">
-      <div className="inner">
-        <LabeledSelect
-          label={getString('release-version')}
-          name="releaseId"
-          value={bundleState.releaseId}
-          onChange={handleVersionChange}>
-          {RELEASES.map(({ id, name }) => (
-            <option key={id} value={id}>
-              {name}
-            </option>
-          ))}
-        </LabeledSelect>
-
-        <LabeledSelect
-          label={getString('language')}
-          name="bundleLocale"
-          value={locale}
-          onChange={handleLangChange}>
-          {sortedLocales.map(locale => (
-            <Localized key={locale} id={locale}>
-              <option value={locale} />
-            </Localized>
-          ))}
-        </LabeledSelect>
+      <div className="dataset-corpus-download-container">
+        <div className="table-text">
+          <PageHeading>
+            <Localized id="download-dataset-header" />
+          </PageHeading>
+          <Localized
+            id="download-delta-explainer"
+            elems={{
+              deltaLink: <DeltaReadMoreLink className="link" />,
+            }}>
+            <div />
+          </Localized>
+          <p style={{ marginTop: '2rem' }}>
+            <Localized id="download-dataset-tag" />
+          </p>
+        </div>
+        <div className="input-row">
+          <LabeledSelect
+            label={getString('language')}
+            name="bundleLocale"
+            value={locale}
+            onChange={handleLangChange}>
+            {useSortedLocales(
+              languagesWithDatasets.map(s => s.name),
+              getString
+            )[0].map(val => (
```

Comment:
The result from `useSortedLocales(...)` should be stored in a variable and mapped here.

```
const sortedLocales = useSortedLocales(...);

<div>
{sortedLocales.map(...)}
</div>
```